instrumentation/logging/logger.go

```diff
@@ -0,0 +1,182 @@
+package logging
+
+import (
+	"fmt"
+	"io"
+	stdlog "log"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+
+	"go.undefinedlabs.com/scopeagent/tags"
+)
+
+const (
+	log_Regex_Template = `(?m)^%s(?:(?P<date>\d{4}\/\d{1,2}\/\d{1,2}) )?(?:(?P<time>\d{1,2}:\d{1,2}:\d{1,2}(?:.\d{1,6})?) )?(?:(?:(?P<file>[\w\-. \/\\:]+):(?P<line>\d+)): )?(.*)\n?$`
+)
+
+type (
+	otWriter struct {
+		logRecordsMutex sync.RWMutex
+		logRecords      []opentracing.LogRecord
+		regex           *regexp.Regexp
+		timeLayout      string
+	}
+	logItem struct {
+		time       time.Time
+		file       string
+		lineNumber string
+		message    string
+	}
+	loggerPatchInfo struct {
+		current  io.Writer
+		previous io.Writer
+	}
+)
+
+var (
+	patchedLoggersMutex sync.Mutex
+	patchedLoggers      = map[*stdlog.Logger]loggerPatchInfo{}
+	stdLoggerWriter     io.Writer
+)
+
+// Patch the standard logger
+func PatchStandardLogger() {
+	stdLoggerWriter := getStdLoggerWriter()
+	otWriter := newInstrumentedWriter(stdlog.Prefix(), stdlog.Flags())
+	stdlog.SetOutput(io.MultiWriter(stdLoggerWriter, otWriter))
+	recorders = append(recorders, otWriter)
+}
+
+// Unpatch the standard logger
+func UnpatchStandardLogger() {
+	stdlog.SetOutput(stdLoggerWriter)
+}
```

Comment:
because we already have `PatchLogger(logger *stdlog.Logger)`

Reply:
That's what I mean. Can we just treat the standard logger as a regular `*stdlog.Logger` and reuse those functions?

Reply:
we don't have the `*stdlog.Logger` instance, it's private at package level. So for that case I used the functions without instance, like `stdlog.SetOutput` instead of using `logger.SetOutput`.

Reply:
Uhm ok